src-tauri/src/permissions.rs

```diff
@@ -0,0 +1,65 @@
+use Permission::*;
+use serde_json::{to_string_pretty, Value};
+use std::{fs::File, io::{Read, Write}};
+
+#[derive(Clone)]
+pub enum Permission {
+    Clipboard,
+    Location,
+    Microphone,
+    Screenshot,
+    Tts
+}
+
+impl Permission {
+    pub fn as_str(&self) -> &str {
+        match *self {
+            Clipboard => "Clipboard",
+            Location => "Location",
+            Microphone => "Microphone",
+            Screenshot => "Screenshot",
+            Tts => "Tts"
+        }
+    }
+}
+
+pub fn update(permissions: Value) {
+    let pretty_json = to_string_pretty(&permissions).unwrap();
+    let mut file = File::create("permissions.json").expect("Failed to create permissions.json!");
+    file.write_all(pretty_json.as_bytes()).expect("Failed to write to file");
+}
+
+fn get() -> Value {
+    let mut file = File::open("permissions.json").expect("Failed to open permissions.json!");
+    let mut json_string = String::new();
+    file.read_to_string(&mut json_string).expect("Failed to read permissions.json!");
+    let permissions: Value = serde_json::from_str(&json_string).expect("Failed to parse permissions.json!");
+    permissions
+}
+
+pub fn check(required: Vec<Permission>) -> Option<String> {
+    let permissions = get();
```

Comment:
kinda nit picky, but maybe it should be called get_permissions or something like that? Since its a private function maybe it doesn't really matter

Reply:
fixed it to get_permissions